File: src/leetcode/L704.java
package leetcode;

// Binary Search
public class L704 {
    public static int search(int[] nums, int target) {
        int left = 0, right = nums.length-1;
        while(left <= right) {
            int pivot = (right + left) / 2;
            if (target == nums[pivot]) return pivot;
            if (target > nums[pivot]) left = pivot+1;
            if (target < nums[pivot]) right = pivot-1;
        }
        return -1;
    }

    public static void main(String[] args) {
        System.out.println(search(new int[]{-1,0,3,5,9,12}, 9));
        System.out.println(search(new int[]{-1,0,3,5,9,12}, 2));
    }
}

File: src/leetcode/L704.py
from typing import List


class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        while l <= r:
            mid = (l + r) // 2
            print(l, r, mid)
            if nums[mid] == target:
                return mid
            if target > nums[mid]:
                l = mid + 1
            elif target < nums[mid]:
                r = mid - 1
        return -1

s = Solution()
print(s.search([5], 5))
